connector/dialect.tdd

```diff
@@ -379,61 +138,34 @@
       <formula>CAST(%1 AS TIMESTAMP)</formula>
       <argument type='datetime' />
     </function>
-    <function group='cast' name='FLOAT' return-type='real'>
+    <function group='cast' name='STR' return-type='str'>
       <formula>(CASE
-	WHEN %1 THEN 1.0
-	WHEN NOT %1 THEN 0.0
-	ELSE NULL END)</formula>
+        WHEN %1 THEN &apos;1&apos;
+        WHEN NOT %1 THEN &apos;0&apos;
```

Comment:
changed to `true`, `false`

Reply:
So in general, does tableau use the `formula` in SQL queries as opposed to the actual function?

If that is the case, I wonder if we should instead be using our `CAST` function instead of this `CASE` formula. `CASE` statements lead to pretty big translations, whereas a `CAST` is smaller and more straightforward.

Reply:
Ah yes, it uses what is in `formula` for that part of the query, for example here's the calculated field input where `Enrolled` is a boolean column:
```
STR([Enrolled])
```
The statement that Tableau runs is:
```
   SELECT (CASE
           WHEN "class"."enrolled" THEN 'true'
           WHEN NOT "class"."enrolled" THEN 'false'
           ELSE NULL END) AS "Calculation_4639622423314558980"
   FROM "class"
```

Reply:
Ok so do we think that's better than `SELECT CAST("class"."enrolled" AS STRING) FROM "class"`? (Or even more succinctly as `SELECT "class"."enrolled"::STRING FROM "class"`? I guess ultimately what I'm asking is, why are we not using the MongoSQL `CAST` function in the formula here?

Reply:
Oh, I see what you mean.  Yes that is more straightforward to use the CAST function here, I made the change.  Also changed the existing as CHAR to as STRING as I see that's how it gets rewritten.